src/view.cpp

```diff
@@ -208,11 +208,17 @@ int View::run()
 			// so
 
 			fa->get_form()->run(-1);
-			fa->process_op(
-				macrocmds[0].op, true, &macrocmds[0].args);
+			if (fa->process_op(
+					macrocmds[0].op, true, &macrocmds[0].args)) {
 
-			// remove first macro command, since it has already been processed
-			macrocmds.erase(macrocmds.begin());
+				// Operation failed, abort
+				macrocmds.clear();
+				show_error(strprintf::fmt(_("Macro execution failed!")));
```

Comment:
Two notes here:
1. I don't think `strprintf::fmt` is necessary, since there is no format specifiers and `_` already returns an appropriate type.
2. Please run `make extract && make msgmerge` to update the POT and PO files with this new string.

Reply:
I ended up removing this message, after fixing the bug you mentioned the browser error was actually displayed, so I think it's better to leave it there so there's a hint on what failed.